radix-engine-stores/src/rocks_db.rs

```diff
@@ -62,6 +62,25 @@ impl CommittableSubstateDatabase for RocksdbSubstateStore {
     }
 }
 
+impl ListableSubstateDatabase for RocksdbSubstateStore {
+    fn list_partition_keys(&self) -> Box<dyn Iterator<Item = DbPartitionKey> + '_> {
+        let mut partition_keys: Vec<DbPartitionKey> = self
+            .db
+            .iterator(IteratorMode::Start)
+            .map(|kv| {
+                let (iter_key_bytes, _) = kv.as_ref().unwrap();
+                let (iter_key, _) = decode_from_rocksdb_bytes(iter_key_bytes);
+                iter_key
+            })
+            .collect();
+        // Eliminate duplicated keys (if any). Rocksdb iterator returns sorted entries, so we can
+        // dedup() directly.
+        partition_keys.dedup();
```

Comment:
I know this impl is just for test purposes, so we don't need to worry about e.g. runtime memory consumption... but still I'd really recommend putting the `.dedup()` in the iterator chain (right before `.collect()`).